public/icvar.h

```diff
@@ -62,13 +62,14 @@ abstract_class ICvar : public IAppSystem
 	virtual CVarDLLIdentifier_t AllocateDLLIdentifier() = 0;
 
 	// Register, unregister commands
-	virtual void			RegisterConCommand( ConCommandBase *pCommandBase, bool unknown=true ) = 0;
+	virtual void			RegisterConCommand( ConCommandBase *pCommandBase, bool cmdline=true ) = 0;
 	virtual void			UnregisterConCommand( ConCommandBase *pCommandBase ) = 0;
 	virtual void			UnregisterConCommands( CVarDLLIdentifier_t id ) = 0;
 
 	// If there is a +<varname> <value> on the command line, this returns the value.
 	// Otherwise, it returns NULL.
-	virtual const char*		GetCommandLineValue( const char *pVariableName ) = 0;
+	inline static const char *GetCommandLineValue( const char *pVariableName );
+	virtual bool			HasCommandLineValue( const char *pVariableName ) = 0;
 

```

Comment:
I like the idea of the shim. Although while it can be static, I think it makes sense for it not to be, to keep the usage the same since the whole reason of it is for compatibility.